### DISCUSSION TITLE
Is that possible to run job which takes more than 6 hours on self-hosted runner?

### DISCUSSION THREAD

**Discussion by victorx-deckard** [community/community#26679] [Actions]
<p>In my CI/CD setup, one step is to retrain a deep learning model. We setup our own GPU-box as a self-hosted runner to run this step. However, the training takes about one day, which is much longer than the current 6-hours cap GitHub Actions allowed. Is there any way to workaround this limitation? </p>

**Comment by Yanjingzhu** [comment]
<p>Yes, you can use self-hosted runner to run more than 6 hours job.</p>
<ul>
<li>Self-hosted runners give you the opportunity to persist whatever you like for your jobs and not be subject to the six-hour time-out in hosted virtual environments.</li>
</ul>
<p>Please refer to this official blog:</p>
<p><a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.blog%2F2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta%2F%23advantages-of-self-hosted-runners&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C44909bba12774497cebc08d77957db12%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637111290551403160&amp;sdata=%2BbXcIk2KVtTU%2FHK%2F8k%2Br2l8Gzuvq3dTynCmNiSAcMBc%3D&amp;reserved=0" rel="nofollow noopener">https://github.blog/2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta/#advantages-of-self-hosted-runners</a></p>

**Comment by victorx-deckard** [comment]
<p>This is interesting. My test shows the jos is killed after 6 hours even running on self-hosted runner. </p>
<p>Here is a snapshot of the summary of the run. As it shows, the job was killed after 6 hours.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/469bfe75d25e38cc0b2388d35f5982a5a75d51cd.png" data-download-href="/uploads/short-url/a4DDAphAkm6TgZFrcyaraOMvCe9.png?dl=1" title="Screenshot from 2019-12-06 11-03-48.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/51894961/181099172-a8c81ef0-b4b8-4bd6-9b42-4cfbe93bc184.png" alt="Screenshot from 2019-12-06 11-03-48.png" data-base62-sha1="a4DDAphAkm6TgZFrcyaraOMvCe9" width="690" height="168" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/469bfe75d25e38cc0b2388d35f5982a5a75d51cd_2_690x168.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/469bfe75d25e38cc0b2388d35f5982a5a75d51cd_2_1035x252.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/469bfe75d25e38cc0b2388d35f5982a5a75d51cd_2_1380x336.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/469bfe75d25e38cc0b2388d35f5982a5a75d51cd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2019-12-06 11-03-48.png</span><span class="informations">1475×361 32.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Just in case one might suspect the job had not been run on self-hosted runner. Here is the screenshot of the shell where I run the self-hosted runner agent. As can be seen from the timestamps, the job (the 3rd and 4th ones) were killed after 6 hours.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/edee0bfd4eb9b988208f528077aea292da4c5f15.png" data-download-href="/uploads/short-url/xWPaKSytPNwWZurH7IeBj36M94V.png?dl=1" title="Screenshot from 2019-12-06 11-05-01.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/51894961/181099173-4b562e31-c7ed-4ec0-99ab-b500ecd74712.png" alt="Screenshot from 2019-12-06 11-05-01.png" data-base62-sha1="xWPaKSytPNwWZurH7IeBj36M94V" width="690" height="230" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/edee0bfd4eb9b988208f528077aea292da4c5f15_2_690x230.png, https://aws1.discourse-cdn.com/github/original/2X/e/edee0bfd4eb9b988208f528077aea292da4c5f15.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/e/edee0bfd4eb9b988208f528077aea292da4c5f15.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/edee0bfd4eb9b988208f528077aea292da4c5f15_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2019-12-06 11-05-01.png</span><span class="informations">987×330 62.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>The maximum number of minutes to let a job run before GitHub automatically cancels it. In default is 360 minutes. </p>
<p>Please try to specify a larger minutes to timeout-minutes under your job. </p>
<p><a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes</a></p>

**Comment by Jianlin-lv** [comment]
<p>I was very confused. It didn’t work for me.  </p>
<p>Even “Both jobs. &lt;job_id&gt; .steps.timeout-minutes” and “jobs. &lt;job_id&gt; .timeout-minutes” be set to 7200,  The jobs was still canceled by GitHub automatically.</p>
<p>My YAML file:</p>
<p><a href="https://github.com/Jianlin-lv/buildx-demo/blob/master/.github/workflows/build_container.yml" rel="nofollow noopener">https://github.com/Jianlin-lv/buildx-demo/blob/master/.github/workflows/build_container.yml</a></p>
<p><a href="https://github.com/Jianlin-lv/buildx-demo/actions/runs/105486327" rel="nofollow noopener">https://github.com/Jianlin-lv/buildx-demo/actions/runs/105486327</a></p>

**Comment by DavidGOrtega** [comment]
<p>Solution is false.<br>
Jobs can not run longer than six hours by any meaning</p>

**Comment by ashipanw** [comment]
<p>+1 . Can some one please update if we can run any jobs more than 6hrs ?</p>

**Comment by ashipanw** [comment]
<p>Just to let all the folks know, Yes its possible to run job/steps for more than 6 hrs , provided you are using self hosted runners .for ex:  Use job-timeout: 1000 parameter to allow jobs/steps to run for 1000 mins in your workflow.</p>
<p>Cheers !!</p>

**Comment by acevos01** [comment]
<p>Can someone help with this? This solution doesn’t work. I’m still getting a 6h timeout even after putting timeout-minutes in the step and job level with a self hosted runner. As aside note I’m using reusable workflows (reusable workflows can’t have timeout-minutes property)</p>

**Comment by ptc-aramachandra** [comment]
Do we have any updates on the solution. Self hosted runners even with timeout parameter being more than 6 hours doesnt work. It still cancels the job after 6 hours. Any solutions here ?

**Comment by schr4der** [comment]
I'm running into the same issue with a self-hosted runner.

**Comment by tt-rkim** [comment]
Still running into this. Any movement on allowing longer than 360 min jobs on self hosted runners? We would like to run stress tests on custom hardware, which shouldn't affect extra runner infrastructure GitHub makes for Actions.

**Comment by XargonWan** [comment]
I can report that still the limit is in place and I'm unable to do a longer build beyond 6h.

**Comment by XargonWan** [comment]
I can report that still the limit is in place and I'm unable to do a longer build beyond 6h.

**Reply by JohnDalyIonQ** [reply]
Thanks for saying something about this, it's an extremely frustrating problem to hit, especially if you have to wait 6 hours to see if you fixed it!

**Comment by tt-rkim** [comment]
We were actually able to get longer than 6h builds on self-hosted runners recently with something like the following:

```
jobs:
  stress-build-and-unit-tests:
    timeout-minutes: 1440
    runs-on: "self-hosted"
    steps:
      - ...
```

The key seems to be set to it at the job level, not step level

**Reply by chintanp** [reply]
@tt-rkim what is the maximum timeout value have you been able to test this for? One of the limits is 35 days, I wonder if this is able to get to that or cross that limit.

**Reply by tt-rkim** [reply]
We have been able to get up to 25 hours, unfortunately no more than 35 days.

I am unsure how to get around the total workflow time... unfortunately my suspicion is because the workflow is a GitHub-managed construct in their architecture or whatever, this doesn't sound like something you could get around. Or at least, I have no ideas currently on how to .
